tool/vfx/vfxSection.h

```diff
@@ -124,67 +124,80 @@ enum ShaderType {
   SpirvAsmFile, // SPIRV assemble code in external file
 };
 
-// =====================================================================================================================
-// Gets offset of the member
-// Clang has a warning for using offsetof for non-POD type
-#define OFFSETOF(T, name) ((size_t)(&((T *)(0))->name))
+template <typename T, T> struct GetMemberHelper;
+
+template <typename T, typename R, R(T::*member)> struct GetMemberHelper<R(T::*), member> {
+  static void *getMemberPtr(void *obj) {
+    T *t = static_cast<T *>(obj);
+    return &(t->*member);
+  }
+};
+
+template <typename T2, typename T, T> struct GetSubStateMemberHelper;
+
+template <typename T2, typename T, typename R, R(T::*member)> struct GetSubStateMemberHelper<T2, R(T::*), member> {
+  static void *getMemberPtr(void *obj) {
+    T2 *t = static_cast<T2 *>(obj);
+    return &(t->getSubStateRef().*member);
+  }
+};
 
 // =====================================================================================================================
 // Initiates a member to address table
-#define INIT_MEMBER_NAME_TO_ADDR(T, name, type, _isObject)                                                             \
+#define INIT_MEMBER_NAME_TO_ADDR(T, name, type, memType, _isObject)                                                    \
   tableItem->memberName = STRING(name);                                                                                \
   if (!strncmp(tableItem->memberName, "m_", 2))                                                                        \
     tableItem->memberName += 2;                                                                                        \
-  tableItem->memberOffset = (unsigned)(OFFSETOF(T, name));                                                             \
-  tableItem->memberType = type;                                                                                        \
+  tableItem->getMember = GetMemberHelper<type T::*, &T::name>::getMemberPtr;                                           \
```

Comment:
How about just `tableItem->getMember = GetMemberHelper<decltype(&T::name), &T::name>::getMemberPtr;`? This way you don't need both `type` and `memType` at the same time.

Reply:
Seems to work, thanks, I’ll update this tomorrow.
I wondered before if `auto` could help but I couldn’t fit it in there ;)

Reply:
Yeah, in C++17 you could just do
```
template <auto Member>
struct GetMemberHelper {
```
and be done with this.